invenio_records_resources/references/entity_resolvers/base.py

```diff
@@ -66,11 +72,13 @@ def _parse_ref_dict(self):
 
     def _parse_ref_dict_type(self):
         """Parse the TYPE from the reference dict."""
-        return _parse_ref_dict(self._ref_dict)[0]
+        r = _parse_ref_dict(self._ref_dict)
+        return r[0] if r else None
```

Comment:
minor: it might be easier to just fail early in these cases, since I'm not sure if consumers of these methods take into account that they might return `None`. If `self._ref_dict` is not a `dict` maybe it's better to catch that in the constructor.

Reply:
This was also just added because of the dict inheritance of record classes, which others might not have. So it can be removed, when the other issue is resolved